[pkg/ottl] A new OTTL function `ParseInt` (#40786)
open-telemetry/opentelemetry-collector-contrib: pkg/ottl/ottlfuncs/README.md

- `ParseCSV("\"555-555-5556,Joe Smith\",joe.smith@example.com", "phone,name,email", mode="ignoreQuotes")`

### ParseInt

`ParseInt(target, base)`

The `ParseInt` Converter interprets a string `target` in the given `base` (0, 2 to 36) and returns its integer representation.

`target` is the string to be converted. `target` should be a valid integer represented in string format. For example, "1234" is a valid `target` value, but "notANumber" is not. The `target` may begin with a leading sign: "+" or "-".

`base` is an `int64` representing the base of the number in the `target` string. An error occurs if the `base` argument is a negative integer.

> If the `base` argument is 0, the true base is implied by the string's prefix following the sign (if present): 2 for "0b", 8 for "0" or "0o", 16 for "0x", and 10 otherwise. Also, for `base` value is 0 only, underscore characters are permitted as defined by the Go syntax for [integer literals](https://go.dev/ref/spec#Integer_literals).

Reviewer: ```suggestion
If the `base` argument is 0, the true base is implied by the string's prefix following the sign (if present): 2 for "0b", 8 for "0" or "0o", 16 for "0x", and 10 otherwise. When the `base` value is 0, underscore characters are permitted as defined by the Go syntax for [integer literals](https://go.dev/ref/spec#Integer_literals).
```
Wording.

Examples of `ParseInt` behavior when `base` is 0: 
- `ParseInt("0b1111_0000", 0) -> 240`
- `ParseInt("0b10110", 0) -> 22`
- `ParseInt("0xFF", 0) -> 255`
- `ParseInt("-0xFF", 0) -> -255`
- `ParseInt("-0o123", 0) -> -83`

The return type is `int64`.

> For more information, please refer to the golang [strconv.ParseInt](https://pkg.go.dev/strconv#ParseInt) documentation.

Reviewer: ```suggestion
For more information, please refer to the documentation for the Go [strconv.ParseInt](https://pkg.go.dev/strconv#ParseInt) function.
```

- `ParseXML("<HostInfo hostname=\"example.com\" zone=\"east-1\" cloudprovider=\"aws\" />")`

### ParseInt

`ParseInt(target, base)`

The `ParseInt` Converter converts a string to its integer representation. 

> `target` is the string to be converted.

Author: Done

> `base` is an `int64` representing the base of the number in the string `target`.

Author: Done
